coral-component-dialog/src/scripts/Dialog.js

```diff
@@ -189,6 +189,9 @@ const Dialog = Decorator(class extends BaseOverlay(BaseComponent(HTMLElement)) {
       tagName: 'coral-dialog-header',
       insert: function (header) {
         header.classList.add(`${CLASSNAME}-title`);
+        // Providing the ARIA attributes to coral dialog header
+        header.setAttribute('role', 'heading');
+        header.setAttribute('aria-level', '2');
```

Comment:
Hi @deepak9798 ,
The dialog is a common component used at various location, it is not necessary it will contain a text. Please confirm this is desired in all cases.

Reply:
Sure !
I will check it. Hopefully, it is desired in all cases. But still dig it out once again.

Reply:
@Pareesh 
I checked it, it is desired in all the cases. If dialog did not have header text, then "coral-dialog-header" component is not generated on GUI.

Please merge this PR. So,  I will procced with granite-ui-coral-spectrum package json change and e2e for the same